### PR TITLE
implement findByDesignation with supporting tests

### PR DESCRIPTION
**Pull request by frett**

### CODE REVIEW

**Comment by frett** [comment]
@hlbraddock

File: api/src/main/java/org/ccci/idm/user/dao/UserDao.java [hunk 1]
```diff
@@ -129,6 +130,16 @@
     List<User> findAllByGroup(@Nonnull Group group, boolean includeDeactivated) throws DaoException;
 
     /**
+     * Find the user with the specified Designation
+     *
+     * @param designation        designation being searched for.
+     * @param includeDeactivated If <tt>true</tt> then deactivated accounts are included.
+     * @return Requested {@link User} or <tt>null</tt> if not found.
+     */
+    @Nullable
```
Comment (hlbraddock) [inline api/src/main/java/org/ccci/idm/user/dao/UserDao.java]: Any reason why `findByDesignation` and `designation` are `@Nullable` whereas `findByEmployeeId` is not?

Comment (frett) [inline api/src/main/java/org/ccci/idm/user/dao/UserDao.java]: I wasn't applying the annotations at the time we implemented `findByEmployeeId` and I like avoiding unnecessary/unrelated code changes when committing or submitting PRs

Comment (hlbraddock) [inline api/src/main/java/org/ccci/idm/user/dao/UserDao.java]: I understand.

**Comment by hlbraddock** [comment]
+1
